addons/utm/models/utm_source.py

```diff
@@ -17,10 +18,14 @@ class UtmSource(models.Model):
 
     @api.model_create_multi
     def create(self, vals_list):
-        new_names = self.env['utm.mixin']._get_unique_names(self._name, [vals.get('name') for vals in vals_list])
+        seach_domain = expression.OR([[('name', '=ilike', name['name'].lower())] for name in vals_list])
+        existing_records, new_vals_list = self.env['utm.mixin']._find_or_create_lower_record('utm.source', vals_list, seach_domain)
```

Comment:
Same here then, if we're doing the same expression.OR twice and are using a new method, might as well do that in the method itself.

Reply:
@reth-odoo what i meant was: what if i am have different domain (instead of =ilike?) i think having domain outside gives much more flexibility